lib/toolkit/loader.js

```diff
@@ -785,7 +786,20 @@ function Loader(options) {
     if (isNative && !uri)
       uri = id;
     let module = Module(id, uri);
-    module.exports = freeze(modules[id]);
+
+    // Lazily expose built-in modules in order to
+    // allow them to be loaded lazily.
+    let cachedModules = modules;
```

Comment:
This code is very confusing! I had to open full file and it took me few passes to grok it. So if I do understand it correctly now you capture passed modules into `cachedModules` just before it `modules` is set to this new object in order to allow lazy access. If that so here are my suggestion:

No need to copy original `modules` in each iteration, specially in the callback since it's not obvious what `modules` is at this point. I would suggest to create `const builtinModuleExports = modules` on the line above `modules = ...`.